tests/components/lock/test_init.py

```diff
@@ -104,10 +104,6 @@ async def test_lock_open_with_code(hass: HomeAssistant) -> None:
 
     assert lock.state_attributes == {"code_format": r"^\d{4}$"}
 
-    with pytest.raises(ValueError):
-        await _async_open(lock, ServiceCall(DOMAIN, SERVICE_OPEN, {}))
-    with pytest.raises(ValueError):
-        await _async_open(lock, ServiceCall(DOMAIN, SERVICE_OPEN, {ATTR_CODE: ""}))
```

Comment:
Can we still the cases with the empty code and assert the validation will be skipped?

Reply:
That's a better proposal. I will adjust it tomorrow.